canon_derive/src/lib.rs

```diff
@@ -57,7 +57,9 @@ fn add_store_bound(mut generics: Generics) -> Generics {
         paren_token: None,
     });
     let mut bounds = Punctuated::new();
-    bounds.push(store_trait);
+
+    // insert at beginning, to not trip up const generics
+    bounds.insert(0, store_trait);
```

Comment:
I don't think this would work, the `store_trait` are just types, so if the structure we derive contains lifetimes we'll got the same error. Lifetimes needs to be first, then types, then consts.

Reply:
You cannot derive Canon for types with lifetimes anyway.